test/blackbox-tests/test-cases/watching/fs-memo.t

```diff
@@ -272,3 +276,159 @@ then creating a file.
   Updating path_stat cache for "file-1": Skipped
   Updating path_stat cache for "file-5": Skipped
   Updating path_stat cache for "file-5": Skipped
+
+Demonstrate that watching symbolic links doesn't currently work.
+
+First, create a symbolic link.
+
+  $ test "ln -s ../file-3 dir/file-6"
+  ------------------------------------------
+  Success, waiting for filesystem changes...
+  353
+  Success, waiting for filesystem changes...
+  ------------------------------------------
+  Updating dir_contents cache for "dir": Updated { changed = true }
+  Updating dir_contents cache for "dir/file-6": Skipped
+  Updating file_digest cache for "dir": Skipped
+  Updating file_digest cache for "dir/file-6": Skipped
+  Updating path_stat cache for "dir": Updated { changed = false }
+  Updating path_stat cache for "dir/file-6": Skipped
+
+Now, delete the symbolic link. Dune receives the corresponding events but
+doesn't rerun the affected action.
+
+# CR-someday amokhov: Fix this.
```

Comment:
As discussed offline, the behaviour is correct here: Dune is just restoring the action from the cache. Anonymous actions are just cached for longer that normal rules.

Reply:
Thanks for spotting that this test was actually working correctly! I've updated the testsuite, and now `fs-memo` has only one issue with symbolic links.

Reply:
When we fix a problem by fixing the test :tada: